tests/hlsl-intrinsic/wave-mask/wave.slang

```diff
@@ -37,25 +37,34 @@ int exclusivePrefixSum(WaveMask mask, int index, int waveLaneId, int originalVal
         
         // Write to shared memory
         sharedMem[index] = val;
-        
-        // Syncronizes on the mask, and ensures memory fence for shared data write
-        WaveMaskSharedSync(localMask); 
         return val;
     }    
     
     return 0;    
 }
 
+// It matters how kernels with WaveMask intrinsics are launched(!). 
+// TODO(JS):
+// If I launch with an numthreads amount that is not the size of the Wave on the device, then some 
+// lanes will not be executing at startup, and the kernel will have to know that is the case. 
+// This works currently though because the mask is only used
+// on CUDA, and it's Wave size is 32.
```

Comment:
We should probably figure out what the best practices would be for developers to compute the "initial" active mask at the start of their program. My hope would be that your `~WaveMask(0)` idiom, or something like it would be something we would *guarantee* will always work.